src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -98,21 +96,9 @@ public FileSystemWatcher(string path) : this(path, "*")
         /// </devdoc>
         public FileSystemWatcher(string path, string filter)
         {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
-
-            // Early check for directory parameter so that an exception can be thrown as early as possible.
-            if (path.Length == 0)
-                throw new ArgumentException(SR.Format(SR.InvalidDirName, path), nameof(path));
-
-            if (!Directory.Exists(path))
-                throw new ArgumentException(SR.Format(SR.InvalidDirName_NotExists, path), nameof(path));
-
+            CheckPathValidity(path);
             _directory = path;
-            _filter = filter ?? throw new ArgumentNullException(nameof(filter));
-
-            if (_filter == "*.*")
-                _filter = "*";
+            Filters.Add(filter); // ?? throw new ArgumentNullException(nameof(filter));
```

Comment:
You need to throw for filter as it used to. Make sure we have tests that check the order of exceptions as well as all of the exceptions (e.g., if both are null you get "path", etc.)